bashflux/bashflux.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"log"
+
+	"github.com/mainflux/mainflux/bashflux/cmd"
+	"github.com/spf13/cobra"
+)
+
+type Config struct {
+	HTTPHost string
+	HTTPPort int
+}
+
+func main() {
+
+	conf := &Config{
+		HTTPHost: "localhost",
+		HTTPPort: 0,
+	}
+
+	// Root
+	var rootCmd = &cobra.Command{
+		Use: "bashflux",
+		PersistentPreRun: func(cmdCobra *cobra.Command, args []string) {
+			// Set HTTP server address
+			cmd.SetServerAddr(conf.HTTPHost, conf.HTTPPort)
+		},
+	}
+
+	// Root Commands
+	rootCmd.AddCommand(cmd.CmdVersion)
+	rootCmd.AddCommand(cmd.CmdClients)
+	rootCmd.AddCommand(cmd.CmdChannels)
+	rootCmd.AddCommand(cmd.CmdMessages)
+	rootCmd.AddCommand(cmd.CmdSession)
+	rootCmd.AddCommand(cmd.CmdUsers)
+
+	// Clients
+	cmd.CmdClients.AddCommand(cmd.CmdCreateClient)
```

Comment:
Please do these initializations as part of the `cmd` package. Moreover, split the `cmd` package into `clients`, `users`, `channels` (for example), and provide a factory method into each of these.